alvr/client_openxr/src/extra_extensions/spatial_marker_tracking.rs

```diff
@@ -0,0 +1,299 @@
+            entity_id_count_output: 0,
+            entity_ids: self.entity_ids.as_mut_ptr(),
+            entity_state_capacity_input: MAX_MARKERS_COUNT as u32,
+            entity_state_count_output: 0,
+            entity_states: self.entity_states.as_mut_ptr(),
+        };
+        unsafe {
+            super::xr_res((self.spatial_entity_fns.query_spatial_component_data)(
+                completion.snapshot,
+                &query_contition,
+                &mut query_result,
+            ))?;
+        }
+        let marker_count = query_result.entity_id_count_output;
+
+        let mut bounded_2d_list = sys::SpatialComponentBounded2DListEXT {
+            ty: sys::SpatialComponentBounded2DListEXT::TYPE,
+            next: ptr::null_mut(),
+            bound_count: marker_count,
+            bounds: self.bounded_2d_arr.as_mut_ptr(),
+        };
+        query_result.next = (&raw mut bounded_2d_list).cast();
+
+        let mut marker_list = sys::SpatialComponentMarkerListEXT {
+            ty: sys::SpatialComponentMarkerListEXT::TYPE,
+            next: ptr::null_mut(),
+            marker_count,
+            markers: self.marker_arr.as_mut_ptr(),
+        };
+        bounded_2d_list.next = (&raw mut marker_list).cast();
+
+        unsafe {
+            super::xr_res((self.spatial_entity_fns.query_spatial_component_data)(
+                completion.snapshot,
+                &query_contition,
+                &mut query_result,
+            ))?;
+        }
+
+        let mut out_markers = vec![];
+        for idx in 0..query_result.entity_id_count_output as usize {
+            if self.entity_states[idx] != sys::SpatialEntityTrackingStateEXT::TRACKING
+                || self.marker_arr[idx].capability != CAPABILITY
+                || self.marker_arr[idx].data.buffer_id == sys::SpatialBufferIdEXT::NULL
+                || self.marker_arr[idx].data.buffer_type != sys::SpatialBufferTypeEXT::STRING
+            {
+                alvr_common::debug!(
+                    "Parsing marker failed! {:?} {:?} {:?} {:?}",
+                    self.entity_states[idx],
+                    self.marker_arr[idx].capability,
+                    self.marker_arr[idx].data.buffer_id,
+                    self.marker_arr[idx].data.buffer_type
+                );
+                continue;
+            }
+
+            let get_info = sys::SpatialBufferGetInfoEXT {
+                ty: sys::SpatialBufferGetInfoEXT::TYPE,
+                next: ptr::null(),
+                buffer_id: self.marker_arr[idx].data.buffer_id,
+            };
+
+            let string = unsafe {
+                let mut _string_lenght = 0;
+                super::xr_res((self.spatial_entity_fns.get_spatial_buffer_string)(
+                    completion.snapshot,
+                    &get_info,
+                    self.string_buffer.len() as u32,
+                    &mut _string_lenght,
+                    self.string_buffer.as_mut_ptr(),
+                ))?;
```

Comment:
This will fail the entire function if one string is oversized and I wasn't able to find anything in the spec that the max size is 25 characters. Since we're turning it into a string anyway I'd suggest to do this:

Call the function with zero input capacity, then it'll write the needed output capacity. Then we can create a vec<u8> with the needed size and then slap that into `CString::from_vec_with_nul` and then call `into_string()`.

The entire chain doesn't do any allocations, but it does do some extra checking so ig it's slower.
And yes, you'll need to cast the vec ptr, but whilst it's somehow not officially documented even the CStr code assumes that `c_char` is compatible with `u8`.